File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left, total, min_sub = 0, 0, float('inf')
        
        for right in range(len(nums)):
            
            total += nums[right]
            
            while total >= target:
                min_sub = min(min_sub, right-left+1)
                total -= nums[left]
                left += 1
                
        return 0 if min_sub == float('inf') else min_sub

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.js
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */
var minSubArrayLen = function(target, nums) {
    let left = 0
    let total = 0
    let min_sub = Infinity
    
    for(let right = 0; right < nums.length; right++){
        total += nums[right]
        
        while(total >= target){
            min_sub = Math.min(min_sub, right-left+1)
            total -= nums[left]
            left += 1
        }
    }
    
    return min_sub === Infinity ? 0 : min_sub
};

